### PR TITLE
Enable error and no_data overrides when testing alerting.

### LINKED ISSUE #7149: [Bug] Test Alert Rule is not testing result handler (where no data & exec error rules are handled)
Please include this information:
- What Grafana version are you using? Master
- What datasource are you using? OpenTSDB
- What OS are you running grafana on? Amazon Linux
- What did you do? Created and tested an alert and set alerting if no datapoints in last 1 min
- What was the expected result? firing = true as there were no datapoints in last 1 min
- What happened instead? firing = false

If it relates to *alerting*
- An image of the test execution data fully expanded.
screen shot 2017-01-05 at 11 31 32 am
screen shot 2017-01-05 at 11 31 39 am
screen shot 2017-01-05 at 11 31 47 am

### PR DESCRIPTION
**Pull request by bergquist**
closes #7149

### CODE REVIEW

File: pkg/services/alerting/eval_handler.go [hunk 1]
```diff
@@ -60,6 +61,40 @@ func (e *DefaultEvalHandler) Eval(context *EvalContext) {
 	context.Firing = firing
 	context.NoDataFound = noDataFound
 	context.EndTime = time.Now()
+	context.Rule.State = e.overrideState(context)
```
Comment (torkelo) [inline pkg/services/alerting/eval_handler.go]: not sure I understand the naming. Reading this as your overriding some behavior or default state? why not e.getNewCurrentState() or determineNewState()

Comment (bergquist) [inline pkg/services/alerting/eval_handler.go]: `e.getNewState()` ?

Comment (torkelo) [inline pkg/services/alerting/eval_handler.go]: yes, that is probably best

**Comment by torkelo** [comment]
looks good!

**Pull request merged**